Jenkinsfile

```diff
@@ -30,6 +30,22 @@ boolean doc_only_change() {
     return rc == 1
 }
 
+boolean release_candidate() {
+    return !sh(label: "Determine if building (a PR of) an RC",
+              script: "git diff-index --name-only HEAD^ | grep -q TAG && " +
+                      "grep -i '[0-9]rc[0-9]' TAG",
+              returnStatus: true)
+}
+
+def scons_faults_args() {
+    // if the faults-enabled pragma is false or it a release candidate; disable fault injection
+    if ((cachedCommitPragma(pragma: 'faults-enabled', def_val: 'true') != 'true') || release_candidate()) {
+        return "BUILD_TYPE=release"
```

Comment:
Per previous discussions about _secure by default_, we shouldn't be forcing this here.  Since a missing `BUILD_TYPE` setting should mean no FI, then we should use that.  It is critically important to test that (_secure by default_) functionality though.  Are we doing this anywhere?

I'm honestly not quite sure what that testing looks like since it's testing that a non-FI build indeed does **not** have FI compiled in, but our current PR and even master landing builds are FI builds.

Reply:
https://jira.hpdd.intel.com/browse/DAOS-5973

Reply:
One suggestion I was given was to see if we could do a weekly run with fault injection disabled.

Reply:
> One suggestion I was given was to see if we could do a weekly run with fault injection disabled.

That was one of the mechanisms that I had considered too.